swervelib/ctre/KrakenX60DriveControllerFactoryBuilder.java

```diff
@@ -0,0 +1,120 @@
+package frc.team670.mustanglib.swervelib.ctre;
+
+import com.ctre.phoenix.motorcontrol.*;
+import com.ctre.phoenix.motorcontrol.can.TalonFXConfiguration;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import frc.team670.mustanglib.swervelib.DriveController;
+import frc.team670.mustanglib.swervelib.DriveControllerFactory;
+import frc.team670.mustanglib.swervelib.ModuleConfiguration;
+
+public final class KrakenX60DriveControllerFactoryBuilder {
+    // REPLACE CONSTANTS
+    private static final double TICKS_PER_ROTATION = 2048.0;
+
+    private static final int CAN_TIMEOUT_MS = 250;
+    private static final int STATUS_FRAME_GENERAL_PERIOD_MS = 250;
+
+    private double nominalVoltage = Double.NaN;
+    private double currentLimit = Double.NaN;
+
+    public KrakenX60DriveControllerFactoryBuilder withVoltageCompensation(double nominalVoltage) {
+        this.nominalVoltage = nominalVoltage;
+        return this;
+    }
+
+    public boolean hasVoltageCompensation() {
+        return Double.isFinite(nominalVoltage);
+    }
+
+    public DriveControllerFactory<ControllerImplementation, Integer> build() {
+        return new FactoryImplementation();
+    }
+
+    public KrakenX60DriveControllerFactoryBuilder withCurrentLimit(double currentLimit) {
+        this.currentLimit = currentLimit;
+        return this;
+    }
+
+    public boolean hasCurrentLimit() {
+        return Double.isFinite(currentLimit);
+    }
+
+    private class FactoryImplementation
+            implements DriveControllerFactory<ControllerImplementation, Integer> {
+        @Override
+        public ControllerImplementation create(Integer id, String canbus,
+                ModuleConfiguration moduleConfiguration) {
+            TalonFXConfiguration motorConfiguration = new TalonFXConfiguration();
+
+            double sensorPositionCoefficient = Math.PI * moduleConfiguration.getWheelDiameter()
+                    * moduleConfiguration.getDriveReduction() / TICKS_PER_ROTATION;
+            double sensorVelocityCoefficient = sensorPositionCoefficient * 10.0;
+
+            if (hasVoltageCompensation()) {
+                motorConfiguration.voltageCompSaturation = nominalVoltage;
+            }
+
+            if (hasCurrentLimit()) {
+                motorConfiguration.supplyCurrLimit.currentLimit = currentLimit;
+                motorConfiguration.supplyCurrLimit.enable = true;
+            }
+
+            WPI_TalonFX motor = new WPI_TalonFX(id, canbus);
+            // CtreUtils.checkCtreError(motor.configAllSettings(motorConfiguration),
+            //         "Failed to configure Falcon 500"); // DROPPED FROM API
```

Comment:
ok. so are all the configurations handled in the rest of this code or is there stuff still left to handle? and why do we need this dead code in here?